src/formatDate.js

```diff
@@ -9,6 +9,74 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  let newDate = '';
+  let fromCharacter = 0;
+  let month;
+  let day;
+  let year;
+  let yearFormat;
+
+  fromFormat.forEach((element) => {
+    const toCharacter = fromCharacter + element.length;
+
+    if (element === 'MM') {
+      month = date.slice(fromCharacter, toCharacter);
+      fromCharacter = fromCharacter + 3;
+    }
+
+    if (element === 'DD') {
+      day = date.slice(fromCharacter, toCharacter);
+      fromCharacter = fromCharacter + 3;
+    }
+
+    if (element === 'YYYY') {
+      year = date.slice(fromCharacter, toCharacter);
+      fromCharacter = fromCharacter + 5;
+      yearFormat = element;
+    }
+
+    if (element === 'YY') {
+      year = date.slice(fromCharacter, toCharacter);
+      fromCharacter = fromCharacter + 3;
```

Comment:
The increment of 'fromCharacter' should also be dynamic here, taking into account the actual separator length.